include/libtorrent/alert_types.hpp

```diff
@@ -1259,44 +1270,45 @@ namespace libtorrent
 		char const* listen_interface() const;
 
 		// the error the system returned
-		error_code error;
+		error_code const error;
 
 		enum op_t
 		{
 			parse_addr, open, bind, listen, get_socket_name, accept, enum_if, bind_to_device
 		};
 
 		// the specific low level operation that failed. See op_t.
-		int operation;
+		int const operation;
 
 		// the type of listen socket this alert refers to.
-		socket_type_t sock_type;
+		socket_type_t const socket_type;
 
 		// the address libtorrent attempted to listen on
 		// see alert's documentation for validity of this value
-		libtorrent::address address;
+		libtorrent::address const address;
 
 		// the port libtorrent attempted to listen on
 		// see alert's documentation for validity of this value
-		int port;
+		int const port;
 
 #ifndef TORRENT_NO_DEPRECATE
 		// the address and port libtorrent attempted to listen on
 		tcp::endpoint endpoint;
+
+		// the type of listen socket this alert refers to.
+		socket_type_t sock_type;
```

Comment:
ah, ok, you got this one. This can't be an enum class though, it would break the API